packages/next/src/index.ts

```diff
@@ -1637,7 +1643,11 @@ export const build: BuildV2 = async buildOptions => {
       const nftCache = Object.create(null);
       const lstatSema = new Sema(25);
       const lstatResults: { [key: string]: ReturnType<typeof lstat> } = {};
-      const pathsToTrace = mergedPageKeys.map(page => pages[page].fsPath);
+      const pathsToTrace = mergedPageKeys
+        .map(page => {
+          return pages[page]?.fsPath;
```

Comment:
Why does `mergedPageKeys` contain non-existent pages, feel like we should address this at the source instead of adding optionals throughout.